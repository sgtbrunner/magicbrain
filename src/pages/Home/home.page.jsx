import React from 'react';
import PropTypes from 'prop-types';

import Rank from '../../components/Rank';
import Logo from '../../components/Logo';
import ImageLinkForm from '../../components/Inputs/ImageLinkForm';
import FaceRecognition from '../../components/FaceRecognition';

const Home = ({ name, entries, imageUrl, box, onInputChange, onImageDetect }) => (
  <div>
    <Rank name={name} entries={entries} />
    <Logo />
    <ImageLinkForm onInputChange={onInputChange} onButtonClick={onImageDetect} />
    <FaceRecognition imageUrl={imageUrl} box={box} />
  </div>
);

Home.propTypes = {
  name: PropTypes.string.isRequired,
  entries: PropTypes.number.isRequired,
  imageUrl: PropTypes.string.isRequired,
  box: PropTypes.shape({
    topRow: PropTypes.number,
    bottomRow: PropTypes.number,
    rightCol: PropTypes.number,
    leftCol: PropTypes.number,
  }).isRequired,
  onInputChange: PropTypes.func.isRequired,
  onImageDetect: PropTypes.func.isRequired,
};

export default Home;
