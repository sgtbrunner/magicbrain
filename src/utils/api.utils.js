const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
