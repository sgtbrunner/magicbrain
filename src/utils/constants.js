export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
