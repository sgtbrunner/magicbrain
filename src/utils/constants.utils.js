const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
