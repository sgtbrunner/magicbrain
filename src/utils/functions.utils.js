export const getUserFromLocalStorage = () => localStorage.getItem('user');
