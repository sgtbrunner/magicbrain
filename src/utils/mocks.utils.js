export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
