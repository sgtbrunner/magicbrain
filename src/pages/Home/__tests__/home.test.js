import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import Home from '..';

const USER_ENTRIES = 'user-entries';
const ENTRIES = 7;

const mountHomePage = () =>
  mountComponent({
    component: Home,
    props: {
      user: {
        id: 3,
        name: 'test',
        email: 'test@test.com',
        entries: ENTRIES,
      },
      setUser: jest.fn(),
    },
  });

describe('Home Page', () => {
  it('should match snapshot', () => {
    const { container } = mountHomePage();

    expect(container).toMatchSnapshot();
  });

  it('should render welcoming text correctly', () => {
    mountHomePage();

    const text = screen.getByText('Hello test, your current entry count is...');
    expect(text).toBeInTheDocument();
  });

  it('should render entries correctly', () => {
    mountHomePage();

    const entries = screen.getByTestId(USER_ENTRIES);
    expect(entries).toHaveTextContent(ENTRIES);
  });
});
