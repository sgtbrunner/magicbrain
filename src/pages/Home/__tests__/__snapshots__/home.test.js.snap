// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should match snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="white f4"
      >
        Hello test, your current entry count is...
      </div>
      <div
        class="white f2"
        data-testid="user-entries"
      >
        7
      </div>
    </div>
    <div
      class="ma4 mt0 tiltbox centered"
    >
      <div
        class="Tilt shadow-2"
        style="height: 130px; width: 130px;"
      >
        <div
          class="Tilt-inner pa4 tiltbox"
        >
          <img
            alt="logo"
            class="noselect"
            src="brain.png"
          />
        </div>
      </div>
    </div>
    <label
      class="f3"
      for="image-link-form"
    >
      The 
      <b>
        MagicBrain
      </b>
       will detect faces in your pictures. Give it a try!
      <form
        class="pa4 br3 mt3 custom-shadow-w form smaller center"
        data-testid="image-link-form"
        id="image-link-form"
      >
        <input
          class="f4 pa2 w-70 ba"
          placeholder="enter the image URL here"
          type="text"
        />
        <button
          class="w-30 f4 link ph3 pv2 dib light bg-light-purple noselect"
          disabled=""
          type="submit"
        >
          Detect
        </button>
      </form>
    </label>
    
  </div>
</div>
`;
